Example/Backpack/View Controllers/ColorsViewController.swift

```diff
@@ -0,0 +1,101 @@
+            fatalError("ColorsViewController collectionView must be using UICollectionViewFlowLayout")
+        }
+        
+        collectionView?.contentInset = UIEdgeInsets(top: BPKSpacingBase, left: BPKSpacingBase, bottom: BPKSpacingBase, right: BPKSpacingBase)
+        
+        layout.estimatedItemSize = ColorPreviewCollectionViewCell.referenceSize()
+    }
+}
+
+// MARK: datasource UICollectionViewDataSource
+
+extension ColorsViewController {
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return  ColorsViewController.allColors.count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return ColorsViewController.allColors[section].count
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ColorsViewController.cellIdentifier, for: indexPath) as? ColorPreviewCollectionViewCell
+            else {
+                fatalError("No cell registered for reuse with identifier \(ColorsViewController.cellIdentifier)")
+        }
+        
+        let (name, color) = ColorsViewController.allColors[indexPath.section][indexPath.row]
+        cell.name = name
+        cell.color = color
+        
+        return cell
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        if (kind == UICollectionElementKindSectionHeader) {
+            guard let headerView = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: ColorsViewController.headerIdentifier, for: indexPath) as? ColorPreviewCollectionViewHeader else{
+                fatalError("Color View Headers are expected to be of type ColorPreviewCollectionViewHeader")
+            }
+            // Customize headerView here
+            switch(indexPath.section){
+            case 0:
+                headerView.name = "Primary colors"
+            case 1:
+                headerView.name = "Secondary colors"
+            case 2:
+                headerView.name = "Grays"
+            default:
+                headerView.name = "No such section"
+            }
+            
+            
+            return headerView
+        }
+        fatalError("Headers are the only supplimental elements available")
+    }
+}
+
+extension ColorsViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+      return ColorPreviewCollectionViewHeader.referenceSize(collectionView: collectionView, text: "Some colors")
```

Comment:
Could we use the actual text for the given section instead of `"Some colors"`? Maybe we can adjust the data structure a bit fit the section header names in there too? Also we should really be using NSLocalizedString for the headers in case we ever whish to localize the app